Pull in sqipu html2pdf updates

rossanderson75 opened the pull request:
Merges in updates from the squipu/html2pdf master repro to match version 5.2.7

HiredMussel committed:
reimplement use of multibyte aware substring when setting newline position

f-houssait committed:
- Added compatibility with PHPUnit 9.0

tom-christensen-mayden committed:
removing builds from 5.1 to 7.2

tom-christensen-mayden committed:
using more PHPUnit/Framework/TestCase in place of PHPUnit_Framework_TestCase and changing assertContains() to assertStringContainsString()

tom-christensen-mayden committed:
fixing tests from upgrading phpunit to ^8.5 and some WIP slight editing to fix warning with phpunit.xml

tom-christensen-mayden committed:
fixing warning about whitelisting in phpunit.xml

tom-christensen-mayden committed:
Delete .phpunit.result.cache

tom-christensen-mayden committed:
now that phpunit has been upgraded, I have set tests to not fail on warning in the xml

elsbury13 committed:
Add PHP 82 codesniffer and travis to run 7.4 up

elsbury13 committed:
Use focal for travis build

elsbury13 committed:
Add libonig5 to before script of travis

elsbury13 committed:
Add 2 protected properties

dynamic properties have been Deprecated in PHP 8.2

elsbury13 committed:
Update .travis.yml

Remove 8.0 from travis

elsbury13 committed:
Update to add compatibility file to git ginore and remove compatibility check not needed

elsbury13 committed:
Remane compatibility.txt to PHPCompatibilityReport.txt

spipu committed:
add support of PHP 8.1 and PHP 8.2

spipu committed:
Merge pull request #733 from jausions/phpunit-9-compatibility

Added compatibility with PHPUnit 9

spipu committed:
update changelog

spipu committed:
better unit tests

spipu committed:
better unit tests

rossanderson75 committed:
Removes duplicate variable declarations

rossanderson75 committed:
fixes syntax error in composer.json from when I was fixing merge conflicts

rossanderson75 committed:
Removes duplicate use statement

rossanderson75 committed:
fixes more conflicts

rossanderson75 committed:
Removes duplicate declarations

rossanderson75 committed:
Adds blank line at end of gitignore file

kyam-mayden commented on .gitignore:
```diff
@@ -17,5 +17,6 @@
 /bin/
 /vendor/
 /composer.lock
+/composer.phar
```
How come we've added this line?

elsbury13 replied:
Removed as this file is not there [692ae07](https://github.com/may-den/html2pdf/pull/3/commits/692ae0768ae548a9f0ac661b4eaf26abce9f8df9)

kyam-mayden commented on phpunit.xml:
```diff
@@ -6,10 +6,24 @@
+    <!-- logging>
+        <log type="coverage-html" target="build/logs/coverage" charset="UTF-8" yui="true" highlight="true"/>
+        <log type="coverage-clover" target="build/logs/clover.xml"/>
+    </logging -->
```
M: Do we need this, or can we delete entirely?

elsbury13 replied:
Dont think we need this, Ive checked a few other phpunit files in other projects and we dont use this
[692ae07](https://github.com/may-den/html2pdf/pull/3/commits/692ae0768ae548a9f0ac661b4eaf26abce9f8df9)

elsbury13 committed:
Remove non used code and file that does not exist

Remove line from gitignore as we do not have that file
Remove commented out code, we do not need this

liamjcrewe commented on composer.json:
```diff
@@ -19,7 +19,7 @@
     },
     "require-dev": {
-        "phpunit/phpunit": "^8.5",
+        "phpunit/phpunit": "^5.0 || ^9.0",
```
🟠  Are we sure this shouldn't just be `^9.0`? I know this came from the main repo but our fork didn't have v5.0 support before

elsbury13 replied:
Yeah we dont need this, it looks like its only there to support 5.6 in there main. Updated to 9.6 latest for PHP7.4 support
[7edccbb](https://github.com/may-den/html2pdf/pull/3/commits/7edccbb26ee4b0858d71fd3ac0210db11b33e141)

liamjcrewe commented on phpunit.xml:
```diff
@@ -1,14 +1,16 @@
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-     xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/8.5/phpunit.xsd"
-     bootstrap="./vendor/autoload.php" colors="true" backupGlobals="false"
+     xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.1/phpunit.xsd"
```
🟠 How come we downgraded from 8.5 to 4.1?

elsbury13 replied:
🤷 changed it to match the PHPUnit version
[7edccbb](https://github.com/may-den/html2pdf/pull/3/commits/7edccbb26ee4b0858d71fd3ac0210db11b33e141)

elsbury13 committed:
Update to use PHPUnit 9.6 add tests to travis and update phpunit.xml
